li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaFederatedProducerImpl.java

```diff
@@ -328,13 +131,40 @@ public void flush() {
 
   @Override
   public void flush(long timeout, TimeUnit timeUnit) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    if (_producers.isEmpty()) {
+      LOG.warn("No producers to flush for cluster group {}", _clusterGroup);
+      return;
+    }
+
+    LOG.info("Flushing LiKafkaProducer for cluster group {} in {} {}...", _clusterGroup, timeout, timeUnit);
+
+    long startTimeMs = System.currentTimeMillis();
+    long deadlineTimeMs = startTimeMs + timeUnit.toMillis(timeout);
```

Comment:
Ah.. flush is also doing the same thing with MAX_VALUE ?

Reply:
See the above comment.